src/ApacheOrcDotNet.OptimizedReader/Infrastructure/StreamDetail.cs

```diff
@@ -0,0 +1,10 @@
+﻿using ApacheOrcDotNet.Protocol;
+
+namespace ApacheOrcDotNet.OptimizedReader.Infrastructure
+{
+    public record StreamDetail(int StreamId, int ColumnId, long FileOffset, int Length, StreamKind StreamKind, ColumnEncodingKind EncodingKind, int DictionarySize)
+    {
+        public StreamPositions Positions { get; init; }
+        public StreamRange Range { get; set; }
```

Comment:
What if you just include `public StreamRange LastRange {get;set}` here rather than storing that in some other datatype?